mender-update/cli/cli.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-update/cli/cli.hpp>
+
+#include <iostream>
+
+#include <common/conf.hpp>
+
+#include <mender-update/context.hpp>
+#include <mender-update/cli/actions.hpp>
+
+namespace mender {
+namespace update {
+namespace cli {
+
+namespace conf = mender::common::conf;
+
+ExpectedAction ParseUpdateArguments(
+	vector<string>::const_iterator start, vector<string>::const_iterator end) {
+	if (start == end) {
+		return expected::unexpected(conf::MakeError(conf::InvalidOptionsError, "Need an action"));
+	}
+
+	if (start[0] == "show-artifact") {
+		unordered_set<string> options {};
+		conf::CmdlineOptionsIterator iter(start + 1, end, options, options);
```

Comment:
Because both option sets without arguments (`--option`) and with arguments (`--option=value`) are empty, so we can pass the same variable to both. If any options existed, we wouldn't be able to do it this way.